### ISSUE TITLE
Question: Why are both RLS and rust-analyzer being used together?

### ISSUE THREAD

**Issue by qdrs** [Shadorain/shadovim#4]
I have noticed that rust-tools requires rust-analyzer to work properly, however most rust guides for neovim mention that you only need to use rust-tools (which comes with analyzer), and that you shouldn't use RLS. For example here : https://sharksforarms.dev/posts/neovim-rust/ .

However if I disable RLS in shadow.vim the autocomplete doesn't work correctly, it doesn't show me the available methods on an object for example.

So in summary I have 2 questions:

- Why are you using both together, is it cause of the same issue I am encountering if I don't use RLS?
- Have you attempted to do it with just rust-analyzer, if so what did you encounter?
Thanks

**Comment by Shadorain** [comment]
Hmm.... thank you for bringing this up. I originally was going to just use rust-analyzer (because of rust-tools) but I found that paired with RLS I got some good lsp diagnostics and whatever else. 

I actually had the same problem when trying to remove RLS so I am assuming that RLS is controlling the rust source for nvim-cmp (completion) so I just kept RLS because I get good completions. Rust-analyzer might provide better ones though, that is a good point, I should look into that more. 

If you find anything more let me know as well 😁 
Thanks!

**Comment by qdrs** [comment]
I think I made some progress, disable procMacro for rust-analyzer in your rust-tools settings, and also remove rust-analyzer in your LSP initliaization as rust-tools adds it in anyway, let me know if that works for you ?

**Comment by simrat39** [comment]
Yeah def remove rust analyzer from [here](https://github.com/Shadorain/shadovim/blob/master/lua/lsp_config.lua#L197) and remove [this whole block](https://github.com/Shadorain/shadovim/blob/master/lua/lsp_config.lua#L206-L212) and it should be fine

**Comment by Shadorain** [comment]
> Yeah def remove rust analyzer from [here](https://github.com/Shadorain/shadovim/blob/master/lua/lsp_config.lua#L197) and remove [this whole block](https://github.com/Shadorain/shadovim/blob/master/lua/lsp_config.lua#L206-L212) and it should be fine

Oh dang, you made rust-tools.nvim didn't ya? I'm assuming something else would handle the rust_analyzer lsp initialization then if I'm removing it from there?

**Comment by Shadorain** [comment]
> I think I made some progress, disable procMacro for rust-analyzer in your rust-tools settings, and also remove rust-analyzer in your LSP initliaization as rust-tools adds it in anyway, let me know if that works for you ?

Ah I see, you are right, rust-tools does it for me, don't need it twice. That is smart. I'll try that!

**Comment by Shadorain** [comment]
Thank you guys, this helped a lot!
```
commit 8aa90a94bcdddde7fd17a21892c3fba1d71c7f5a (HEAD -> master, origin/master, origin/HEAD)
Author: Shadorain <shadow@Shadowraith>
Date:   Mon Nov 8 10:41:10 2021 -0500

    #4: removed RLS now just use rust-tools.nvim which auto initializes rust-analyzer; faster and cleaner
```

**Comment by simrat39** [comment]
I'm pretty sure you don't need to disable proc macros as well.
Also you put it in the tool opts where it does nothing

**Comment by qdrs** [comment]
> I'm pretty sure you don't need to disable proc macros as well. Also you put it in the tool opts where it does nothing

The latest rust update broke proc_macros in rust-analyzer, try it out !

**Comment by Shadorain** [comment]
> I'm pretty sure you don't need to disable proc macros as well. Also you put it in the tool opts where it does nothing

True, didn't notice this will update
